webpack/scenes/ContentViews/Details/Versions/ContentViewVersions.js

```diff
@@ -38,29 +42,64 @@ const ContentViewVersions = ({ cvId }) => {
     __('Description'),
   ];
 
-
   useDeepCompareEffect(() => {
+    const buildCells = (cvVersion) => {
```

Comment:
You've ended up with an absolutely enormous effect here.  What are your thoughts on using `useCallback` instead? That would allow you to move `buildCells`, `buildActiveTaskCells`, and `buildRows` back outside this effect, and make this file a little more readable.

Having done this several dozen times in my work on the last PR, the procedure I've gotten in the habit of using is

1. Move the function outside the effect. `eslint` will tell you
```
React Hook useDeepCompareEffect has a missing dependency: 'buildCells'. Either include it or remove the dependency array. (react-hooks/exhaustive-deps)
```
2. Do as it says, and add `buildCells` to the array. Now, `eslint` says
```
The 'buildCells' function makes the dependencies of useDeepCompareEffect Hook (at line 116) change on every render. Move it inside the useDeepCompareEffect callback. Alternatively, wrap the definition of 'buildCells' in its own useCallback() Hook. (react-hooks/exhaustive-deps)
```
3. We're going to choose the latter option. Add `useCallback`:
```js
  const buildCells = useCallback((cvVersion) => {
    ...
  });
```
4. Now `eslint` will tell you
```
React Hook useCallback does nothing when called with only one argument. Did you forget to pass an array of dependencies? (react-hooks/exhaustive-deps)
```
5. Add `[]` to the end of the function:
```js
  }, []);
```
Now, `eslint` complains again and tells you exactly what is needed:
```
React Hook useCallback has a missing dependency: 'cvId'. Either include it or remove the dependency array. (react-hooks/exhaustive-deps)
```
6. Add `cvId` and you're good to go!

It's really not too bad once you get used to it ;)

Reply:
> Nothing happens when I close the modal right after starting the task.

Ya..If you close the modal before the publish API returns a task response and the modal displays the loading spinner, nothing happens..Andrew and I were discussing about this and probably giving a warning below Loading like Please do not close or something but it seems unnecessary cause in the worst case if you do close it when loading, the page needs a refresh manually. 

Working on the useCallback.. I was waiting for you to come tell me what to do to avoid making the effect so huge.. :smile:

Reply:
> in the worst case if you do close it when loading, the page needs a refresh manually.

On the Subscriptions page we always show a progress bar if there's a manifest-related task in progress. And it will pop up even if you navigate to the page from somewhere else. 

We could do something similar here, but instead of having a persistent noncloseable progress bar we could just use the polling to automatically refresh the table when the task is done.  Just a thought..

Reply:
The task progress on the table will show up if you navigate from someplace else that involves a refresh of the rows..That happens also when you close the wizard from the task presenter page..

However the case where the modal is closed when the publish call hasn't returned a response is when reloading the table doesn't help cause publish method creates actual version fairly late in the planning phase..And if you make a call to versions API while publish is still in planning, it returns the versions list without the new version that is yet to be created and the task that is yet to be started.